modules/cas_cache/layer_cache_management.c

```diff
@@ -204,11 +238,16 @@ static void _cache_mngt_cache_flush_complete(ocf_cache_t cache, void *priv,
 
 	result = _cache_mngt_async_callee_set_result(context, error);
 
-	if (result == -KCAS_ERR_WAITING_INTERRUPTED)
+	if (context->compl_func)
+		context->compl_func(cache);
```

Comment:
At this point context might be freed. You must either 
1. call compl_func before _cache_mngt_async_callee_set_result or 
2. call compl_func  conditionally if result == -KCAS_ERR_WAITING_INTERRUPTED and perform equivalent operations in caller for the case of  result != -KCAS_ERR_WAITING_INTERRUPTED